lib/fluent/plugin/parser_syslog.rb

```diff
@@ -137,6 +153,91 @@ def parse_plain(text, &block)
 
         yield time, record
       end
+
+      SPLIT_CHAR = ' '.freeze
+      PRI_START_CHAR = '<'.freeze
+
+      def parse_rfc3164(text, &block)
+        pri = nil
+        start = 0
+        if @with_priority
+          if text.start_with?(PRI_START_CHAR)
+            i = text.index('>'.freeze, 1)
+            pri = text.slice(1, i - 1).to_i
+            start = i + 1
+          else
+            yield nil, nil
+            return
+          end
+        end
+
+        # header part
+        diff = 15 # skip Mmm dd hh:mm:ss
+        time_end = text[start + diff]
+        if time_end == SPLIT_CHAR
+          time_str = text.slice(start, diff)
+          start += 16 # time + ' '
+        elsif time_end == '.'.freeze
+          # support subsecond time
+          i = text.index(SPLIT_CHAR, diff)
+          time_str = text.slice(start, i - start)
+          start = i + 1
+        else
+          yield nil, nil
+          return
+        end
+
+        i = text.index(SPLIT_CHAR, start)
+        if i.nil?
+          yield nil, nil
+          return
+        end
+        diff = i - start
+        host = text.slice(start, diff)
+        start += (diff + 1)
+
+        i = text.index(SPLIT_CHAR, start)
+        if i.nil?
+          yield nil, nil
+          return
+        end
+        diff = i - start
+
+        record = {'host' => host}
+        record['pri'] = pri if pri
+
+        # message part
+        msg = if text[i - 1] == ':'.freeze
+                if text[i - 2] == ']'.freeze
+                  j = text.index('['.freeze, start)
+                  record['ident'] = text.slice(start, j - start)
+                  record['pid'] = text.slice(j + 1, i - j - 3) # remove '[' / ']:'
+                else
+                  record['ident'] = text.slice(start, i - start - 1)
+                end
+                text.slice(i + 1, text.bytesize)
+              else
+                if @support_colonless_ident
+                  if text[i - 1] == ']'.freeze
+                    j = text.index('['.freeze, start)
+                    record['ident'] = text.slice(start, j - start)
+                    record['pid'] = text.slice(j + 1, i - j - 2) # remove '[' / ']'
+                  else
+                    record['ident'] = text.slice(start, i - start)
+                  end
+                  text.slice(i + 1, text.bytesize)
+                else
+                  text.slice(i - diff, text.bytesize)
+                end
+              end
+        msg.chomp!
+        record['message'] = msg
+
+        time = @time_parser.parse(time_str.squeeze(SPLIT_CHAR))
```

Comment:
time_str was sliced with size(15), right? why do we need to call String#squeeze?

Reply:
I can't remember the actual product name but users hit the problem with the number of ' '.
regexp version also calls `squeeze` for avoiding this problem.

Reply:
Existing regex does not specify the size of chars. but here specifies the fixed size(15).
So if the character which needs to call `squeeze` is passed to this filter plugin, the parser will fail to parse it in any case, right?.